libs/jsonschema/instance.go

```diff
@@ -129,3 +133,34 @@ func (s *Schema) validatePattern(instance map[string]any) error {
 	}
 	return nil
 }
+
+func (s *Schema) validateConst(instance map[string]any) error {
+	for k, v := range instance {
+		fieldInfo, ok := s.Properties[k]
+		if !ok || fieldInfo.Const == nil {
+			continue
+		}
+		if v != fieldInfo.Const {
+			return fmt.Errorf("expected value of property %s to be %v. Found: %v", k, fieldInfo.Const, v)
+		}
+	}
+	return nil
+}
+
+// Validates that the instance matches at least one of the schemas in anyOf
+// but will also succeed if the property values are omitted.
+// For more information, see https://json-schema.org/understanding-json-schema/reference/combining#anyof.
+func (s *Schema) validateAnyOf(instance map[string]any) error {
+	if s.AnyOf == nil {
```

Comment:
This implies that an empty `anyOf` array (as opposed to null) will always return an error.

1. Is that in line with the spec?
2. Could you add a comment for this case if so?

Reply:
Done. Yeah kind of, the spec prohibits empty `anyOf` declarations.